ratpack-session/src/main/java/ratpack/session/AllowedSessionType.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ratpack.session;
+
+import java.lang.annotation.*;
+
+/**
+ * Indicates that the annotated type is always safe to store in a session.
+ * <p>
+ * The default implementation of {@link SessionTypeFilter} provided by {@link SessionModule}
+ * allows any class that carries this annotation to be used in a session.
+ *
+ * @see SessionModule
+ * @since 1.9
+ */
+@Inherited
```

Comment:
This is a bad idea. I would not suggest making this an `@Inherited` annotation.

Reply:
the annotation is gone entirely.

It turns out that inspecting class annotations causes the class to be initialized. Theoretically, a class could do something malicious in its static initializer.